mempool/mempool.go

```diff
@@ -380,10 +381,11 @@ func (mem *Mempool) notifyTxsAvailable() {
 	}
 }
 
-// ReapMaxBytes reaps transactions from the mempool up to n bytes total.
-// If max is negative, there is no cap on the size of all returned
+// ReapMaxBytesMaxGas reaps transactions from the mempool up to maxBytes bytes total
+// with the condition that the total gasWanted must be less than maxGas.
+// If both maxes are negative, there is no cap on the size of all returned
 // transactions (~ all available transactions).
-func (mem *Mempool) ReapMaxBytes(max int) types.Txs {
+func (mem *Mempool) ReapMaxBytesMaxGas(maxBytes int, maxGas int64) types.Txs {
```

Comment:
I don't think this is a good name for it, but the best alternate name I could think of was "ReapBlock" which didn't seem right. (Since the mempool isn't really aware of blocks at the moment)